### PR TITLE
Add web service

### PR DESCRIPTION
**Pull request by malesh**
Add support for API jobs & fix a bug in worker shutdown.

Also upgraded to go 1.18.

### CODE REVIEW

File: store/store.go [hunk 1]
```diff
@@ -91,6 +91,11 @@ func (s *BasicStore) WaitToDoJob() (chan *gotell.Job, error) {
 	return receiver, nil
 }
 
+func (s *BasicStore) StopWaiting(receiver chan *gotell.Job) {
```
Comment (NathanKirby-FTP) [inline store/store.go:94]: I take it this was the bug you were talking about?

File: sender/sender.go [hunk 2]
```diff
@@ -20,11 +20,14 @@ type BySMS interface {
...
+	To(to string, cc ...string)
+}
+
+type ByApi interface {
+	ByPush
 }
 
 type ByPush interface {
-	UID() string
 	WithBody(body string)
```
Comment (NathanKirby-FTP) [inline sender/sender.go:31]: Did you intend to remove UID from the interface?

Comment (NathanKirby-FTP) [reply sender/sender.go:31]: And take it `To` was moved to ByChat as it is only used by that inherited interface?

File: store/store.go (hunk 1 above)
Comment (malesh) [reply store/store.go:94]: Yes, have a look at TestClosingWorkerDoesntBreakThings

File: sender/sender.go (hunk 2 above)
Comment (malesh) [reply sender/sender.go:31]: Nothing uses the UID.
I moved To up because it didn't make sense for the API interface.

**Review by NathanKirby-FTP — APPROVED** [review]
LGTM

**Review by tantricllama — APPROVED** [review]
LGTM

**Pull request merged**